guides/jfrog-cli-plugins-developer-guide.md

```diff
@@ -99,3 +100,39 @@ To create your own private plugins registry, follow these steps.
 To publish a plugin to the private registry, run the following command, while inside the root of the plugin's sources directory. This command will build the sources of the plugin for all the supported operating systems. All binaries will be uploaded to the configured registry.
 
 ```jf plugin publish the-plugin-name the-plugin-version```
+
+## Testing your plugin before publishing it
+When installing a plugin using the ```jf plugin install``` command, the plugin is downloaded into its own directory under the ```plugins``` directory, which is located under the JFrog CLI home directory. By default, you can find the ```plugins``` directory under ```~/.jfrog/plugins/```.
+So if for example you are developing a plugin named ```my-plugin```, and you'd like to test it with JFrog CLI before publishing it, you'll need to place your plugin's executable, named ```my-plugin```, under the following path - 
+```
+plugins/my-plugin/bin/
+```
+If your plugin also uses [external resources](having-your-plugin-use-external-resources), you should place the resources under the following path - 
+```
+plugins/my-plugin/resources/
+```
+Once the plugin's executable is there, you'll be able to see it is installed by just running ```jf```.
+
+## Having your plugin use external resources
+### General
+In some cases your plugin may need to use external resources. For example, the plugin code may need to run an executable or read from a configuration file. You would therefore want these resources to be packaged together with the plugin, so that when it is installed, these resources are also downloaded and become available for the plugin.
+
+### How can you add resources to your plugin?
+The way to include resources for your plugin, is to simply place them inside a directory named `resources` at the root of the plugin's sources directory. You can create any directory structure inside `resources`.
+When publishing the plugin, the content of the `resources` directory is published alongside the plugin executable. When installing the plugin, the resources are also downloaded.
```

Comment:
maybe we can add one sentence about executables for different OS
"pay attention to include executables for all plugin's relevant os in your resources directory" - something like this?